### PR TITLE
Require GINKGO 1.4.0

### PR DESCRIPTION
**Pull request by gfcas**
Recently I had some problems building dealii+ginkgo correctly. With the help of @pratikvn we found that the ginkgo develop branch was required, which is now available as ginkgo-v1.4.0. So this PR introduces a version check and force at least ginkgo 1.4.0.

In https://github.com/ginkgo-project/ginkgo/issues/867 the ginkgo developers propose to check the version via ``find_package(Ginkgo 1.4.0)``. However, in order to follow the style of the other configure files I added a manual check. Are there any opinions for one of the proposed ways to check the version?

### CODE REVIEW

**Comment by upsj** [comment]
I only just noticed that you provide a FindGINKGO.cmake file. Do you intend to always rely on a custom Find module, or would you be okay with using the GinkgoConfig.cmake that is installed by default (and also found by CMake directly)? This is the recommended way to include Ginkgo, and can be used explicitly with `find_package(Ginkgo 1.4.0 CONFIG)`, which skips the Find modules, potentially with `QUIET` to silence warnings if no GinkgoConfig.cmake could be found.

File: cmake/modules/FindGINKGO.cmake [hunk 1]
```diff
@@ -20,6 +20,7 @@
 #
 #   GINKGO_INCLUDE_DIRS
 #   GINKGO_INTERFACE_LINK_FLAGS
+#   GINGKO_VERSION
```
Comment (marcfehling) [inline cmake/modules/FindGINKGO.cmake:23]:
```suggestion
#   GINKGO_VERSION
```

File: cmake/modules/FindGINKGO.cmake [hunk 2]
```diff
@@ -63,6 +64,13 @@ FOREACH(_library ginkgo ${GINKGO_INTERFACE_LINK_LIBRARIES})
     )
 ENDFOREACH()
 
+#
+# Get ginkgo version number
+#
+IF(Ginkgo_FOUND)
```
Comment (marcfehling) [inline cmake/modules/FindGINKGO.cmake:70]:
```suggestion
IF(GINKGO_FOUND)
```

**Review by marcfehling — COMMENTED** [review]
I've just found minor stuff. I'll let someone do the review who works with ginkgo in deal.II...

File: cmake/modules/FindGINKGO.cmake (hunk 2 above)
Comment (gfcas) [reply cmake/modules/FindGINKGO.cmake:70]: Sorry, but at this point the variable ``GINKGO_FOUND`` does not yet exists. If I see it correctly it is first defined in https://github.com/dealii/dealii/blob/b7a9c8420bfd8667c126878a80ce93833b3b15b8/cmake/macros/macro_deal_ii_package_handle.cmake#L60. Here I adopted the mechanism from https://github.com/dealii/dealii/blob/b7a9c8420bfd8667c126878a80ce93833b3b15b8/cmake/modules/FindBOOST.cmake.

**Comment by gfcas** [comment]
Honestly, I am not an expert in cmake. I just wanted to provide a working check (adopted from already existing ones) to avoid for others the installation of a non-supported version. As an initial check I would maybe leave it in this fashion. Maybe someone wants to improve and unify this later?

**Review by tamiko — APPROVED** [review]

**Comment by tamiko** [comment]
@gfcas Excellent. Thanks a lot for cleaning this up!

**Pull request merged**